ccf_transaction_processor/README.md

```diff
@@ -0,0 +1,208 @@
+
+## Share CCF (TLS) Authentication Keys
+
+CCF uses mutually authenticated TLS channels for transactions. Keys are
+located at `$PDO_HOME/ccf/keys`. The network certificate is
+`networkcert.pem`. User public certificate is `user0_cert.pem` and
+private key is `user0_privk.pem`.  Note that the keys are created as
+part of CCF deployment and are unique to the specific instance of CCF.
+
+In our usage, CCF users are PDO clients, and PDO client authentication
+is implemented within the transaction processor itself. Thus, we do not
+utilize the client authentication feature provided by CCF. However to
+satisfy the CCF's requirement that only authorized CCF users can submit
+transactions to CCF, share `user0_cert.pem` and `user0_privk.pem` with
+all the PDO clients. These two keys and the network certificate
+`networkcert.pem` must be stored under $PDO_LEDGER_KEY_ROOT (as part of
+PDO deployment). The user keys must be renamed as `userccf_cert.pem` and
+`userccf_privk.pem` respectively.
+
+## Test the Deployment with Ping Test
+
+PDO TP contains a simple `ping` rpc that returns success every time it
+is queried. Test the PDO-TP deployment using this rpc. Invoke the
+following commands to issue 100 ping rpcs. The net througput is reported
+at the end of the test.
+
+```bash
+${PDO_SOURCE_ROOT}/ccf_transaction_processor/scripts/ping_test.py
+```
+
+## Generate Ledger Authority Key
+
+Responses to read-transactions include a payload signature, where the
+signature is generated within PDO-TP.  The required signing keys must be
+generated before PDO-TP can be opened up for business from PDO
+clients. This will be done automatically if you start the CCF network
+with the script `${PDO_HOME}/ccf/bin/start_ccf_network.sh`.
+
+Otherwise, you may invoke the following commands to generate and save
+the ledger authority key.
+
+```bash
+${PDO_HOME}/ccf/bin/generate_ledger_authority.py
+${PDO_HOME}/ccf/bin/fetch_ledger_authority.py
+```
+
+If successful, the rpc returns after 1) creating one set of signing keys
+locally within the CCF enclave, and 2) scheduling them for global
+commit. The ledger authority verifying key can be obtained using the
+`get_ledger_verifying_key` rpc. The verifying key is returned only after
+global commit is successful.
+
+The read-payload-signature feature may be used by PDO clients to
+establish offline verifiable proof of transaction commits as desired by
+the PDO smart contract application. Note that for trust purposes, it is
+recommended that any entity that uses the verifying_key gets it directly
+from the CCF service using the `get_ledger_verifying_key` rpc.
+
+## Using CCF ledger using PDO
+
+We highlight some quick details about how PDO clients can use a CCF
+based PDO-TP deployment. The information below can be found at
+[PDO docs](../docs) as well.
+
+1. Set the following environment variables:
+
+```bash
+export PDO_LEDGER_TYPE=ccf
+export PDO_LEDGER_URL=http://ip-address:6006
```

Comment:
Hmm, by default our scripts/settings start the server only on 127.0.0.1, so the instruction here might be confusing ...

Is there no way that CCF might just listen on all interfaces?

If not, you will have to add some text here what you have to do if it is not localhost (e.g., for my test case where ccf was running on host and ehs/test in docker, i've changed cchost.toml to point to the address on the docker network (172.20.0.1), restarted network  and then used that in the test ...)

Reply:
Thank you for the question. By default, ccf is started on HOSTNAME:6006, where HOSTNAME is one of the env variables that must be set before configuring CCF. We also recommend setting HOSTNAME to the ipaddress to avoid issues that arise while converting names to ip.

Reply:
I don't think your new text clarifies much.  Most folks would probably read that as the main (non-localhost) ip address of the host, not necessarily the default 127.0.0.1?

Reply:
Thank you. I changed this to 
```bash
export PDO_LEDGER_TYPE=ccf
export PDO_LEDGER_URL=http://ccf-ip-address:6006
```

Note that ccf-ip-address is the ip-address that was set in the env variable HOSTNAME (see above) during CCF
deployment. 
========================================
HOSTNAME is one of the env that we expect users to set before deploying CCF (we also say that the user should set it to an ip rather than name. This can be set to `127.0.0.1` as well. The env is already in PDO. Setting it to a name like localhost might also work (since there is some conversation within scripts, but I am not sure if this conversation works in all cases, so we recommend to set it explicitly to the ip)